packages/eas-cli/src/project/expoConfig.ts

```diff
@@ -37,21 +41,52 @@ export async function createOrModifyExpoConfigAsync(
   }
 }
 
-function getExpoConfigInternal(
+let wasExpoConfigWarnPrinted = false;
+
+async function getExpoConfigInternalAsync(
   projectDir: string,
   opts: ExpoConfigOptionsInternal = {}
-): ExpoConfig {
+): Promise<ExpoConfig> {
   const originalProcessEnv: NodeJS.ProcessEnv = process.env;
   try {
     process.env = {
       ...process.env,
       ...opts.env,
     };
-    const { exp } = getConfig(projectDir, {
-      skipSDKVersionRequirement: true,
-      ...(opts.isPublicConfig ? { isPublicConfig: true } : {}),
-      ...(opts.skipPlugins ? { skipPlugins: true } : {}),
-    });
+
+    let exp: ExpoConfig;
+    if (isExpoInstalled(projectDir)) {
+      try {
+        const { stdout } = await spawnAsync(
+          'npx',
+          ['expo', 'config', '--json', ...(opts.isPublicConfig ? ['--type', 'public'] : [])],
+
+          {
+            cwd: projectDir,
```

Comment:
do we need to pass `env` into spawnAsync? (we do for the expo-updates CLI). Or does the process.env override on L52 work?

Reply:
Technically the override should work, but I passed envs explicitly either way so we are certain it won't break.